DTAConfig/CustomSettings/CustomSettingFileCheckBox.cs

```diff
@@ -0,0 +1,161 @@
+using ClientCore;
+using ClientGUI;
+using Rampastring.Tools;
+using Rampastring.XNAUI;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace DTAConfig.CustomSettings
+{
+    /// <summary>
+    /// An implementation of a check-box that switches between two sets of files.
+    /// </summary>
+    public class CustomSettingFileCheckBox : XNAClientCheckBox, ICustomSetting
+    {
+        public CustomSettingFileCheckBox(WindowManager windowManager) : base(windowManager) { }
+
+        private List<FileSourceDestinationInfo> enabledFiles = new List<FileSourceDestinationInfo>();
+        private List<FileSourceDestinationInfo> disabledFiles = new List<FileSourceDestinationInfo>();
+
+        private bool EnabledFilesComplete => enabledFiles.All(f => File.Exists(f.SourcePath));
+        private bool DisabledFilesComplete => disabledFiles.All(f => File.Exists(f.SourcePath));
+
+        private bool defaultValue;
+        private bool originalState;
+        private bool restartRequired;
+        private bool checkFilePresence;
+        private bool resetUnselectableItem;
+
+        public override void GetAttributes(IniFile iniFile)
+        {
+            base.GetAttributes(iniFile);
+
+            var section = iniFile.GetSection(Name);
+
+            if (section == null)
+                return;
+
+            int i = 0;
+            while (true)
+            {
+                string fileInfo = section.GetStringValue($"EnabledFile{i}", string.Empty);
+
+                if (string.IsNullOrWhiteSpace(fileInfo))
+                    break;
+
+                string[] parts = fileInfo.Split(',');
+                if (parts.Length < 2)
+                {
+                    Logger.Log($"Invalid CustomSettingFileCheckBox information in {Name}: {fileInfo}");
+                    continue;
+                }
+
+                FileOperationOptions options = default;
+                if (parts.Length >= 3)
+                    Enum.TryParse(parts[2], out options);
+
+                enabledFiles.Add(new FileSourceDestinationInfo(parts[0], parts[1], options));
+
+                i++;
+            }
+
+            i = 0;
+            while (true)
+            {
+                string fileInfo = section.GetStringValue($"DisabledFile{i}", string.Empty);
+
+                if (string.IsNullOrWhiteSpace(fileInfo))
+                    break;
+
+                string[] parts = fileInfo.Split(',');
+                if (parts.Length < 2)
+                {
+                    Logger.Log($"Invalid CustomSettingFileCheckBox information in {Name}: {fileInfo}");
+                    continue;
+                }
+
+                FileOperationOptions options = default;
+                if (parts.Length >= 3)
+                    Enum.TryParse(parts[2], out options);
+
+                disabledFiles.Add(new FileSourceDestinationInfo(parts[0], parts[1], options));
+
+                i++;
+            }
```

Comment:
The code inside these two loops is identical aside from the target list and the name of the key. It should be moved into its own method unless there's a very good reason not to do so.